ArduCopter/ekf_check.cpp

```diff
@@ -264,14 +275,11 @@ void Copter::check_vibration()
     const bool innov_velD_posD_positive = is_positive(vel_innovation.z) && is_positive(pos_innovation.z);
 
     // check if vertical velocity variance is at least 1 (NK4.SV >= 1.0)
-    float position_variance, vel_variance, height_variance, tas_variance;
-    Vector3f mag_variance;
-    if (!ahrs.get_variances(vel_variance, position_variance, height_variance, mag_variance, tas_variance)) {
+    if (!variances_valid) {
         innovation_checks_valid = false;
     }
-
     const bool is_vibration_affected = ahrs.is_vibration_affected();
-    const bool bad_vibe_detected = (innovation_checks_valid && innov_velD_posD_positive && (vel_variance > 1.0f)) || is_vibration_affected;
+    const bool bad_vibe_detected = (innovation_checks_valid && innov_velD_posD_positive && (vel_variance.get() > 1.0f)) || is_vibration_affected;
```

Comment:
For users of EKF3, this change win't have much effect on the detection of a vibration degraded nav due to the check returned by ahrs.is_vibration_affected() so this should be OK for a copter application.

Reply:
ok